Makefile

```
@@ -105,17 +105,44 @@ slow_tests_diffusers: test_installs
 	python -m pytest tests/test_diffusers.py -v -s -k "test_deterministic_image_generation"
 	python -m pytest tests/test_diffusers.py -v -s -k "test_no_"
 
-# Run text-generation non-regression tests
+# Run all text-generation non-regression tests
 slow_tests_text_generation_example: test_installs
 	python -m pip install -r examples/text-generation/requirements_awq.txt
 	BUILD_CUDA_EXT=0 python -m pip install -vvv --no-build-isolation git+https://github.com/HabanaAI/AutoGPTQ.git
 	python -m pip install git+https://github.com/HabanaAI/DeepSpeed.git@1.19.0
 	python -m pytest tests/test_text_generation_example.py tests/test_encoder_decoder.py -v -s --token $(TOKEN)
 
+# Run subset of text-generation non-regression tests that require 1 Gaudi card
+slow_tests_text_generation_example_1x: test_installs
+	python -m pip install -r examples/text-generation/requirements_awq.txt
+	BUILD_CUDA_EXT=0 python -m pip install -vvv --no-build-isolation git+https://github.com/HabanaAI/AutoGPTQ.git
+	python -m pytest tests/test_text_generation_example.py tests/test_encoder_decoder.py -m "(not x2) and (not x4) and (not x8)" -v -s --token $(TOKEN)
+
+# Run subset of text-generation non-regression tests that require 2 Gaudi cards
+slow_tests_text_generation_example_2x: test_installs
+	python -m pip install git+https://github.com/HabanaAI/DeepSpeed.git@1.19.0
+	python -m pytest tests/test_text_generation_example.py -m x2 -v -s --token $(TOKEN)
+
+# Run subset of text-generation non-regression tests that require 4 Gaudi cards
+slow_tests_text_generation_example_4x: test_installs
+	python -m pip install git+https://github.com/HabanaAI/DeepSpeed.git@1.19.0
+	python -m pytest tests/test_text_generation_example.py -m x4 -v -s --token $(TOKEN)
+
+# Run subset of text-generation non-regression tests that require 8 Gaudi cards
+slow_tests_text_generation_example_8x: test_installs
+	python -m pip install git+https://github.com/HabanaAI/DeepSpeed.git@1.19.0
+	python -m pytest tests/test_text_generation_example.py -m x8 -v -s --token $(TOKEN)
+
 # Run image-to-text non-regression tests
 slow_tests_image_to_text_example: test_installs
 	python -m pytest tests/test_image_to_text_example.py -v -s --token $(TOKEN)
 
+slow_tests_image_to_text_example_1x: test_installs
+	python -m pytest tests/test_image_to_text_example.py -v -s -m "not x8" --token $(TOKEN)
```

```suggestion
	python -m pytest tests/test_image_to_text_example.py -m "not x8" -v -s --token $(TOKEN)
```

To ensure consistency, use the -m x -v -s pattern.